### ISSUE TITLE
[Bug]: Linter silently fails on parse error

### ISSUE THREAD

**Issue by cramertj** [aspect-build/rules_lint#408]
### What happened?

`bazelisk lint` with a Python target and `ruff` configured should complain if `ruff.toml` is invalid TOML. When using `ruff` directly, invalid TOML results in an error like:

```
➜  pigweed git:(2432b18cb) ✗ ruff check
ruff failed
  Cause: Failed to parse /usr/local/google/home/cramertj/src/pigweed/.ruff.toml
  Cause: TOML parse error at line 16, column 11
   |
16 | include = '\.pyi?$'
   |           ^^^^^^^^^
invalid type: string "\\.pyi?$", expected a sequence
```

But using `bazelisk lint` silently fails (yet says it has completed successfully!):

```
➜  pigweed git:(2432b18cb) ✗ bazelisk lint //pw_rpc/py/...
2024/10/09 14:46:44 Using unreleased version at commit 1ff5af18d045e8f30a2a0367470db4e8225a785c
2024/10/09 14:46:44 Using unreleased version at commit 1ff5af18d045e8f30a2a0367470db4e8225a785c
INFO: Analyzed 13 targets (0 packages loaded, 0 targets configured).
INFO: Found 13 targets...
INFO: Elapsed time: 0.132s, Critical Path: 0.02s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
```

### Version

Output of `bazel --version`: `aspect 2024.40.2-a7ed65a (with local changes)`
